src/Components/test/E2ETest/Infrastructure/ServerFixtures/ServerFixture.cs

```diff
@@ -52,13 +53,30 @@ protected static void RunInBackgroundThread(Action action)
         {
             var isDone = new ManualResetEvent(false);
 
+            ExceptionDispatchInfo edi = null;
             new Thread(() =>
             {
-                action();
+                try
+                {
+                    action();
+                }
+                catch (Exception ex)
+                {
+                    edi = ExceptionDispatchInfo.Capture(ex);
+                }
+
                 isDone.Set();
             }).Start();
 
-            isDone.WaitOne();
+            if (!isDone.WaitOne(TimeSpan.FromSeconds(10)))
+            {
+                throw new TimeoutException("Timed out waiting for: " + action);
+            }
+
+            if (edi != null)
+            {
+                throw edi.SourceException;
+            }
```

Comment:
This wasn't the cause of any problems, but it will cause the process to fail with a bad error if it ever throws. I saw this while looking for the source of the hang, and wanted to harden it.